dev/io.openliberty.cdi.core_fat/fat/src/io/openliberty/cdi/internal/core/injection/CDIInjectionTest.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.cdi.internal.core.injection;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.cdi.internal.core.FATSuite;
+import io.openliberty.cdi.internal.core.injection.app.CDIInjectionTestServlet;
+
+@RunWith(FATRunner.class)
+public class CDIInjectionTest {
+
+    @TestServlet(contextRoot = "cdiInjection", servlet = CDIInjectionTestServlet.class)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        server = FATSuite.server;
+
+        Package appPackage = CDIInjectionTestServlet.class.getPackage();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "cdiInjection.war")
+                                   .addPackage(appPackage)
+                                   .addAsWebInfResource(appPackage, "beans.xml", "beans.xml");
```

Comment:
I thought we said we don't really want any beans.xml. Can you delete this file?

Reply:
Hmm, I added one because it's required for CDI Lite. I know our implementation doesn't require it but it would be best practise for writing a portable application, right?

Reply:
ignore me as it will cause different behaviours for different versions.

Reply:
As for empty beans.xml behaviour for cdi 4.0, we need to add more tests on that instead.

Reply:
I think from CDI 1.2 onwards, having no beans.xml should always be interpreted as discovery-mode=annotated? It's only in CDI 1.0 and CDI Lite where having no beans.xml implies discovery-mode=none.

An empty beans.xml changed behaviour in 4.0 (from all to annotated).

Reply:
You were right. I confused about empty beans.xml and no beans.xml.

> Hmm, I added one because it's required for CDI Lite. I know our implementation doesn't require it but it would be best practise for writing a portable application, right?

We are not running in CDI Lite. We are on CDI full. I think we need to execute the tests without a beans.xml. Not sure whether you can do it with the current beans.xml and without beans.xml and they should produce the same result.